components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/soaptorest/SOAPOperationBindingTestCase.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.impl.soaptorest;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.soaptorest.util.SOAPOperationBindingUtils;
+import org.wso2.carbon.apimgt.impl.utils.APIMWSDLReader;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ServiceReferenceHolder.class})
+public class SOAPOperationBindingTestCase {
+
+    @Test
+    public void testGetSoapOperationMapping() throws Exception {
+        String url  = "http://ws.cdyne.com/phoneverify/phoneverify.asmx?wsdl";
+        String mapping = SOAPOperationBindingUtils.getSoapOperationMapping(url);
+        Assert.assertTrue("Failed getting soap operation mapping from the WSDL", !mapping.isEmpty());
+    }
+
+    @Test
+    public void testGetWSDLProcessor() throws Exception {
+        String url  = "http://ws.cdyne.com/phoneverify/phoneverify.asmx?wsdl";
+        APIMWSDLReader wsdlReader = new APIMWSDLReader(url);
```

Comment:
using public urls is not a good thing for test cases. Would face test failures if this url is changed or if the server is down. shall we try to make this local? (maybe read for the file)

Reply:
Yes, these URLs must be avoided. These can become a nightmare for release processes when the URLs becomes unavailable time to time.